futures-core/src/task_impl/mod.rs

```diff
@@ -204,9 +204,9 @@ impl<T: ?Sized> Spawn<T> {
                                  notify: &N,
                                  id: usize) -> Poll<T::Item, T::Error>
         where N: Clone + Into<NotifyHandle>,
-              T: Future,
+              T: FutureMove,
     {
-        self.poll_notify(notify, id, |s| s.poll())
+        self.poll_notify(notify, id, |s| s.poll_move())
```

Comment:
I don't know enough about the task system. Does it need to be bound `FutureMove`, or is it safe to bound it `Future`. Can we make changes to it, and then it will support `Future`?